addons/binding/org.openhab.binding.homematic/src/main/java/org/openhab/binding/homematic/internal/model/HmDevice.java

```diff
@@ -38,34 +38,30 @@
 
     private List<HmChannel> channels = new ArrayList<HmChannel>();
 
+    public HmDevice(String address, HmInterface hmInterface, String type, String gatewayId, String homegearId,
```

Comment:
A constructor with many strings such as this is error prone (too easy to mess up the order of values), I'm in doubt whether this is really a good alternative, wdyt?

Reply:
You mean a good alternative over the setters? If yes, I disagree: the reason why I replaced the setters with the constructor is that it wasn't clear to me at the beginning what class members are 'static' (only set initially and never touched again afterwards) and what members are expected to change during the object's lifetime. By removing the setters and replacing them by constructor parameters, this immediately becomes clear (albeit I probably should make the member variables final now to make it even more clear - will do that once we reach consensus).

One way to improve (at least a bit) would be reordering the parameter list, e.g. putting `hmInterface` between `type` and `gatewayId`.

Reply:
An alternative which makes your change way bigger is adding real types / enums for `HmDeviceType type, GatewayId gatewayId, HomeGearId homegearId`

If you make sure you validate by hand the refactored code and do some proper testing I'm okay if @gerrieg is okay with this.

Reply:
> An alternative which makes your change way bigger is adding real types / enums for HmDeviceType type, GatewayId gatewayId, HomeGearId homegearId

Unfortunately that doesn't really work. The gateway ID and Homegear ID are just that - (opaque) IDs: gateway ID is the thing ID of the gateway this device is connected to, and Homegear ID is the ID used by Homegear to reference this device. Both can't be meaningfully interpreted. For `type` one could create an enum, however, doing so has disadvantages: type is often used as part of unique ID generation (e.g. for things, thing channels etc.). One could still do this with an enum, but there's a problem when devices emerge that were unknown at binding creation time: one would need to summarize them in an 'unknown' enum value, so the identification function would be lost for those devices, which in all other aspects are likely to work just fine.

If this change is controversial, I can also take it out of this PR as it's not directly related - it was something I just noticed when I made myself familiar with the code base, and other classes (e.g. `HmDatapoint` or `HmDatapointInfo`) already followed the same style.

Reply:
No as long as @gerrieg still approved he already he did, so without any further messages I assume he still does :-) Its okay I just noted that the new choice has its own drawbacks imho. For me it's not a reason to stop this from being merged.